__tests__/proxy-ws-test.ts

```diff
@@ -0,0 +1,168 @@
+import WebSocket from "ws";
+
+import createServer from "./helpers/createServer";
+import MockServerController from "./helpers/mockServerController";
+
+const generalDelay = 10;
+
+interface IWebsocketOptions {
+  openCallback?: (evt: Event) => void;
+}
+function openWSConnection(
+  port,
+  options: IWebsocketOptions = {}
+): Promise<WebSocket> {
+  return new Promise(resolve => {
+    const ws = new WebSocket(`ws://localhost:${port}`);
+    ws.on("open", evt => {
+      if (options.openCallback) {
+        options.openCallback(evt);
+      }
+      resolve(ws);
+    });
+  });
+}
+
+function wait(timeout = 10) {
+  return new Promise(resolve => {
+    setTimeout(resolve, timeout);
+  });
+}
+
+describe("Proxy - XHR", () => {
+  let s;
+  let msPort;
+  let msController;
+
+  beforeEach(async () => {
+    s = await createServer();
+    msController = new MockServerController(s.port);
+    msPort = await msController.start();
+    // msPort = s.port;
+  });
+
+  afterEach(() => {
+    msController.stop();
+    s.server.close();
+  });
+
+  describe("Requests", () => {
+    it("opens a connection", async () => {
+      const mockConnection = jest.fn();
+      s.wss.on("connection", mockConnection);
+
+      const ws = await openWSConnection(msPort);
+      expect(mockConnection).toHaveBeenCalled();
+      ws.close();
+    });
+
+    it("closes a connection", async () => {
+      const mockClose = jest.fn();
+      s.wss.on("connection", (ws: WebSocket) => {
+        ws.on("close", mockClose);
+      });
+
+      const wsClient = await openWSConnection(msPort);
+      wsClient.close();
+      await wait();
```

Comment:
there is no other way? Still not sure if we want to wait for a specified time instead of listening to an event :(

Reply:
I think our openWSConnection could receive a number saying how many messages it expects before resolving a promise. We need to race this promise with a timeout, so if any of the two happen we resolve, because we dont want to wait forever. Although it looks like await on just already has a timeout. So maybe just counting the messages might do.